fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/BankSearchService.java

```diff
@@ -6,19 +6,24 @@
 import de.adorsys.opba.fintech.impl.service.mapper.Mapper;
 import de.adorsys.opba.tpp.bankserach.api.model.BankSearchResponse;
 import de.adorsys.opba.tpp.bankserach.api.resource.TppBankSearchApi;
+import lombok.Setter;
 import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Configuration;
 
 import java.util.stream.Collectors;
 
 @Configuration
 @Slf4j
+@Setter
 public class BankSearchService {
 
+    @Autowired
```

Comment:
I'd propose to make injected beans `final` and use `@RequiredArgsConstructor`.

Then we don't need to use `@Setter` and `@Autowired` anymore, since Spring automatically injects beans in constructor.

Reply:
I did not know how to change service with mocked Instance in test. Thats why I have setter. Its used in FinTechServerTests
        bankSearchService.setTppBankSearchApi(mockedTppBankSearchApi);

Reply:
@MockBean did the job, thanks

Reply:
With final beans it also works - you get constructor with all necessary parameters, so that you can be flexible in tests with mocks

Reply:
and made it final. thx